### PR TITLE
bump gatewayapi to 1.1.0-rc2

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/envoyproxy/gateway/issues/3265

Key features:
- Bump GRPCRoute to `v1`
- Rename `gwapiv1a2.PolicyTargetReferenceWithSectionName` to `gwapiv1a2.LocalPolicyTargetReferenceWithSectionName`

### CODE REVIEW

**Review by AliceProxy — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

File: internal/cmd/egctl/testdata/translate/out/default-resources.all.yaml [hunk 1]
```diff
@@ -322,6 +322,7 @@ grpcRoutes:
       - method:
           method: DoThing
           service: com.example.Things
+      sessionPersistence: null
```
Comment (arkodg) [inline internal/cmd/egctl/testdata/translate/out/default-resources.all.yaml:325]: should be fixed in v1.1
https://github.com/kubernetes-sigs/gateway-api/pull/3059

File: internal/envoygateway/scheme.go [hunk 2]
```diff
@@ -6,11 +6,11 @@
 package envoygateway
 
 import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
-	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
-	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
```
Comment (arkodg) [inline internal/envoygateway/scheme.go:12]: I dont think we can remove these yet
we need to continue to support GRPCRoute in v1beta1

File: internal/gatewayapi/backendtrafficpolicy.go [hunk 3]
```diff
@@ -199,16 +198,11 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 }
 
 func resolveBTPolicyGatewayTargetRef(policy *egv1a1.BackendTrafficPolicy, gateways map[types.NamespacedName]*policyGatewayTargetContext) (*GatewayContext, *status.PolicyResolveError) {
-	targetNs := policy.Spec.TargetRef.Namespace
```
Comment (arkodg) [inline internal/gatewayapi/backendtrafficpolicy.go:202]: can we rm this ? what happens when the target Ns is different f

**Review by arkodg — APPROVED** [review]
LGTM thanks ! 
this bump does not look easy 

once merged, we should probably alert our users on Slack that old BackendTLSPolicy resources will no longer work, and will need to be updated

File: internal/gatewayapi/backendtrafficpolicy.go (hunk 3 above)
Comment (arkodg) [reply internal/gatewayapi/backendtrafficpolicy.go:202]: nvm, this field no longer exists

**Pull request merged**